### PR TITLE
Improve the performance in nearest_index

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
Improves performance of `nearest_index`

Fixes # (issue)

**How Has This Been Tested?**
CI

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by rem1776 — APPROVED** [review]

File: axis_utils/include/axis_utils2.inc [hunk 1]
```diff
@@ -306,39 +306,45 @@
...
+            call mpp_error(FATAL, 'axis_utils2::nearest_index array is NOT monotonously ordered')
+      END DO
+    endif
+
     if (array(1) < array(ia)) then
     !< increasing array
 
       !< Check if the rval is outside the range of the array
-      if (rval < array(1)) then
+      if (rval .le. array(1)) then
```
Comment (bensonr) [inline axis_utils/include/axis_utils2.inc:334]: You changed the `<` here to `.le.` here, but didn't change it below in the _decreasing_ block, though I think you want `.lt.` there.

File: axis_utils/include/axis_utils2.inc [hunk 2]
```diff
@@ -306,39 +306,45 @@
...
          NEAREST_INDEX_ = 1
          return
-      elseif (rval > array(ia)) then
+      elseif (rval .gt. array(ia)) then
```
Comment (bensonr) [inline axis_utils/include/axis_utils2.inc:337]: Same comment with `>` going to `.gt.` as above.

File: axis_utils/include/axis_utils2.inc [hunk 3]
```diff
@@ -306,39 +306,45 @@
...
-               return
-            ENDIF
-            EXIT
-         END IF
+      DO i = 1, ia
+         if (rval .le. array(i)) then
+            NEAREST_INDEX_ = i
+            if (array(i) -rval .gt. rval - array(i-1)) NEAREST_INDEX_ = i - 1
+            return
+         endif
```
Comment (bensonr) [inline axis_utils/include/axis_utils2.inc:347]: Do you need to make a similar change in the _decreasing_ block?

**Comment by uramirez8707** [comment]
@bensonr I think i addressed your comments. Let me know if the logic doesn't seem correct.

**Review by bensonr — COMMENTED** [review]
Make sure the unit tests checks an array and rvals less than zero and data with a mix of positive and negative elements.

**Review by rem1776 — APPROVED** [review]

**Pull request merged**